server/ai_process.go

```diff
@@ -311,8 +312,11 @@ func processAIRequest(ctx context.Context, params aiRequestParams, req interface
 
 	var resp *worker.ImageResponse
 
+	cctx, cancel := context.WithTimeout(context.Background(), processingRetryTimeout)
```

Comment:
Nit: You'll want to pass `ctx` (param value for `processAIRequest()`) as the parent context here to ensure that if there is a cancellation further up the call stack that it properly propagates - no point in continuing here if no one is waiting for the result.